### PR TITLE
fix #259 partially converting some of the components from headlessui to shadcn

### LINKED ISSUE #259: Replace @headlessui/react with all shadcn/ui library components for better efficiency
## Description

Currently, papermark project utilizes the `@headlessui/react` library for Dropdown and Disclosure components across various UI elements. However, we have an alternative library, `shadcn/ui`, already integrated into the project, providing similar functionality with no additional installation requirements. This enhancement proposes replacing `@headlessui/react` components with their counterparts from `shadcn/ui` to **streamline the codebase, reduce application size, and declutter dependencies**.

## Benefits

- **Reduced Application Size:** By eliminating the dependency on `@headlessui/react`, we can reduce the overall size of the application bundle.
- **Project Simplification:** Utilizing components from a single UI library simplifies maintenance and reduces the cognitive load for developers.
- **Consistency:** Ensuring consistency across UI components enhances user experience and improves code maintainability.
- **Optimized Performance:** With reduced dependencies, the application's performance may see marginal improvements, especially in load times and rendering efficiency.

### PR DESCRIPTION
**Pull request by doppedheart**
fix #259 
change the faq from disclore to accordian 
Screenshot 2024-07-22 203158
and also the dropdownproto 
Screenshot 2024-07-22 203153
Screenshot 2024-07-22 203126

### CODE REVIEW

File: app/(static)/deck/components/DropdownProto.tsx
Comment (mfts) [inline app/(static)/deck/components/DropdownProto.tsx:1]: This should be a `select` component from shadcn not a popover.

File: app/(static)/deck/components/DropdownProto.tsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
-import { Menu, Transition } from "@headlessui/react";
+import { Button } from "@/components/ui/button";
 import { CheckIcon, ChevronDownIcon, ChevronUpIcon } from "lucide-react";
-import { Fragment, Dispatch, SetStateAction } from "react";
+import { Dispatch, SetStateAction, useState } from "react";
+import { PopoverContent } from "@radix-ui/react-popover";
+import { PopoverTrigger, Popover } from "@/components/ui/popover";
 
 function classNames(...classes: string[]) {
```
Comment (mfts) [inline app/(static)/deck/components/DropdownProto.tsx]: If you want you can replace this with `cn` from `@/lib/utils`. it's a better way to merge tailwind classes then this helper function.

**Review by mfts — CHANGES_REQUESTED** [review]

File: app/(static)/deck/components/DropdownProto.tsx
Comment (doppedheart) [reply app/(static)/deck/components/DropdownProto.tsx:1]: okay will make the changes

File: app/(static)/deck/components/DropdownProto.tsx (hunk 1 above)
Comment (doppedheart) [reply app/(static)/deck/components/DropdownProto.tsx:7]: there is same code written in two different files 
DropdownProto.tsx -> app/(static)/deck/components/DropdownProto.tsx
dropdownpropt.tsx -> components/web/alternatives/dropdownproto.tsx

should i make the same changes as removing the headlessui components with shadcn in both 
or remove any one file and make the corresponding file import changes related to other

**Comment by doppedheart** [comment]
there is same code written in two different files
DropdownProto.tsx -> app/(static)/deck/components/DropdownProto.tsx
dropdownpropt.tsx -> components/web/alternatives/dropdownproto.tsx

should i make the same changes as removing the headlessui components with shadcn in both
or remove any one file and make the corresponding file import changes related to other

**Comment by doppedheart** [comment]
other than these ui changes there are two more files which contain headlessui , 
1. pages/ai.tsx
2. pages/share-notion-page.tsx
both contain the same Faq component (disclosure as dependency) should i import the component over there as that would be much more cleaner approch,
or else do the same as Faq component but it would be code repetition

**Comment by mfts** [comment]
> 1. pages/ai.tsx
> 2. pages/share-notion-page.tsx
>    both contain the same Faq component (disclosure as dependency) should i import the component over there as that would be much more cleaner approch,

yes that would be preferred! Thank you for spotting that and suggesting this change :) 


Please also merge in the current main branch so that merging becomes easier.

**Comment by doppedheart** [comment]
made changes to all the files and completely removed headlessui
Screenshot 2024-07-27 025342
Screenshot 2024-07-27 025016
Screenshot 2024-07-27 025418

File: app/(static)/deck/components/DropdownProto.tsx [hunk 2]
```diff
@@ -17,58 +20,48 @@ export default function DropDown({
...
+    }
+
+    // If the menu is closed, reset the isFirstClick state
+    if (!open) {
+      setIsFirstClick(false);
+      setMenuOpen(false); // Ensure the dropdown is closed
+    } else {
+      setMenuOpen(true); // Open the dropdown
+    }
+  };
```
Comment (mfts) [inline app/(static)/deck/components/DropdownProto.tsx]: This is not needed. This is only to for a double-click delete confirmation

File: app/(static)/deck/components/DropdownProto.tsx [hunk 3]
```diff
@@ -17,58 +20,48 @@ export default function DropDown({
...
+  return (
+    <Select open={menuOpen} onOpenChange={handleMenuStateChange}>
```
Comment (mfts) [inline app/(static)/deck/components/DropdownProto.tsx]:
```suggestion
    <Select>
```

File: components/ui/accordion.tsx
Comment (mfts) [inline components/ui/accordion.tsx:1]: No need to change the accordion behaviour. Let's leave the chevron. This component is used in other places and relies on the chevron.

File: pages/ai.tsx [hunk 4]
```diff
@@ -67,50 +67,6 @@ const features = [
...
-      "You can use one of the starting commands and get the summary of received docuement. If you want that documents shared with you.",
-  },
-  {
-    question: "How to turn Pitch Deck into Memo?",
-    answer:
-      "You can use Papermark AI and ask it to create Memo from Pitch Deck received. You can also uplaod your own Pitch deck there.",
-  },
-
-  // More questions...
-];
```
Comment (mfts) [inline pages/ai.tsx]: You are removing the questions. These are different from the one in the `@/components/web/faq` component. Ideally we make a FAQ component that takes in props with different questions and answers.

File: components/web/alternatives/dropdownproto.tsx [hunk 5]
```diff
@@ -18,58 +20,48 @@ export default function DropDown({
...
+    }
+
+    // If the menu is closed, reset the isFirstClick state
+    if (!open) {
+      setIsFirstClick(false);
+      setMenuOpen(false); // Ensure the dropdown is closed
+    } else {
+      setMenuOpen(true); // Open the dropdown
+    }
+  };
```
Comment (mfts) [inline components/web/alternatives/dropdownproto.tsx]: No needed. same as above

File: components/web/alternatives/dropdownproto.tsx [hunk 6]
```diff
@@ -18,58 +20,48 @@ export default function DropDown({
...
+  return (
+    <Select open={menuOpen} onOpenChange={handleMenuStateChange}>
```
Comment (mfts) [inline components/web/alternatives/dropdownproto.tsx]:
```suggestion
    <Select>
```

**Review by mfts — CHANGES_REQUESTED** [review]

**Review by mfts — APPROVED** [review]
Looks great! Thank you for your effort!

**Pull request merged**